### PR TITLE
Docs/automatic policy

### PR DESCRIPTION
**Pull request by valentindrdt**

### CODE REVIEW

File: laravel/index.md [hunk 1]
```diff
@@ -649,25 +649,52 @@ To protect an operation and ensure that only authorized users can access it, sta
 php artisan make:policy BookPolicy --model=Book
 ```
 
-If the standard Laravel conventions are followed, the Policy class is autodetected and used automatically.
-Otherwise, you can use the `policy` property on an operation attribute to explicitly enforce a policy:
+And... That's it. Laravel will automatically detect your new policy and use it when manipulating a Book.
```
Comment (soyuka) [inline laravel/index.md]:
```suggestion
Laravel will automatically detect your new policy and use it when manipulating a Book.
```

File: laravel/security.md [hunk 2]
```diff
@@ -1,26 +1,78 @@
 # Security
-
-API platform is compatible with Laravel [authorization](https://laravel.com/docs/authorization) mechanism. Once a gate is defined, you can specify the policy to use within an operation:
+## Policy
```
Comment (dunglas) [inline laravel/security.md]:
```suggestion

## Policies

```

File: laravel/security.md [hunk 3]
```diff
@@ -1,26 +1,78 @@
...
 use ApiPlatform\Metadata\Patch;
 
-#[Patch(policy: 'update')]
+#[Patch]
 class Book extends Model
 {
 }
 ```
 
+API Platform will detect the operation and map it to a specific method in your policy.
```
Comment (dunglas) [inline laravel/security.md]:
```suggestion
API Platform will detect the operation and map it to a specific method in your policy according to the rules defined in this table:

```

File: laravel/security.md [hunk 4]
```diff
@@ -1,26 +1,78 @@
...
+
+private const POLICY_METHODS = [
+    Put::class => 'update',
+    Post::class => 'create',
+    Get::class => 'view',
+    GetCollection::class => 'viewAny',
+    Delete::class => 'delete',
+    Patch::class => 'update',
+];
+```
```
Comment (dunglas) [inline laravel/security.md]:
```suggestion
| Operation      | Policy                                                     |
|----------------|------------------------------------------------------------|
| GET collection | `viewAny`                                                  |
| GET            | `view`                                                     |
| POST           | `create`                                                   |
| PATCH          | `update`                                                   |
| DELETE         | `delete`                                                   |
| PUT            | `update` or `create` if the resource doesn't already exist |
```

File: laravel/security.md [hunk 5]
```diff
@@ -1,26 +1,78 @@
...
+
+If your policy methods do not match Laravel's standards, you can always use the `policy` property on an operation attribute to enforce this policy:
```
Comment (dunglas) [inline laravel/security.md]:
```suggestion
If your policy methods do not match Laravel's conventions, you can always use the `policy` property on an operation attribute to enforce this policy:
```

File: laravel/security.md [hunk 6]
```diff
@@ -1,26 +1,78 @@
...
+ use ApiPlatform\Metadata\ApiResource;
++use ApiPlatform\Metadata\Patch;
+ use Illuminate\Database\Eloquent\Model;
+
+-#[ApiResource]
+ #[ApiResource(
+     paginationItemsPerPage: 10,
++    operations: [
++       new Patch(
++            policy: 'updateBook',
```
Comment (dunglas) [inline laravel/security.md]:
```suggestion
+            policy: 'myCustomPolicy',
```

For the sake of explicitness :)

File: laravel/security.md [hunk 7]
```diff
@@ -1,26 +1,78 @@
...
++       ),
++    ],
+)]
+ class Book extends Model
+ {
+ }
+```
+
+You also can link a model to a policy:
```
Comment (dunglas) [inline laravel/security.md:52]:
```suggestion
You also can link a model to a policy:

```

File: laravel/security.md [hunk 8]
```diff
@@ -1,26 +1,78 @@
...
+```php
+use App\Models\Book;
+use App\Tests\Book\BookPolicy;
+use Illuminate\Support\Facades\Gate;
+
+Gate::guessPolicyNamesUsing(function (string $modelClass) {
```
Comment (dunglas) [inline laravel/security.md]:
```suggestion
Gate::guessPolicyNamesUsing(function (string $modelClass): ?string {
```

**Review by dunglas — CHANGES_REQUESTED** [review]

**Pull request merged**